include/fp_util/memory.h

```diff
@@ -0,0 +1,23 @@
+#ifndef MEMORY_H
+#define MEMORY_H
+namespace fp_util {
+inline void use_memory(void* mem){
+  asm volatile("" : "+g"(mem) : :);
+}
+
+inline void *get_memory(size_t Size){
+  void* mem = malloc(Size);
+
+  use_memory(mem);
+
+  return mem;
+}
+
+inline void free_memory(void* mem){
+  use_memory(mem);

```

Comment:
I'm not sure about this. Why do you think we need this?

Reply:
Not sure if I had a specific intent in mind with this, other than to have both a function for allocation and freeing memory that automatically "uses" the memory. In hindsight, I don't see any realistic use cases for this at the moment to be honest.

I guess we remove it then?